src/inmanta/deploy/scheduler.py

```diff
@@ -559,24 +575,23 @@ async def is_agent_running(self, name: str) -> bool:
 
     # TaskManager interface
 
-    async def get_resource_intent(
-        self, resource: ResourceIdStr, *, for_deploy: bool = False
-    ) -> Optional[tuple[int, ResourceDetails]]:
+    async def get_resource_intent(self, resource_id: ResourceIdStr, *, for_deploy: bool = False) -> Optional[ResourceIntent]:
```

Comment:
I'd prefer to keep this `resource`, again for consistency with other methods.

Reply:
@sanderr style and performance wise, I'm a bit apprehensive about this: 
1. we changed what this method does (it now also checks the state of dependencies)
2. it already had two modes


I wonder if we shouldn't split/copy this in two distinct methods, one to get the resource intent (for get_fact) and one to start deploying?

Reply:
https://inmanta.slack.com/archives/CKRF0C8R3/p1731414114134429